build.cmd

```diff
@@ -1,3 +1,10 @@
 @echo off
+set CACHEDMSBUILDDEBUGONSTART=%MSBUILDDEBUGONSTART%
+if %MSBUILDDEBUGONSTART% == 0 goto build
+set /p ans=Did you mean to have MSBUILDDEBUGONSTART equal %MSBUILDDEBUGONSTART%? (y/n) 
+if %ans% == y goto build
+set MSBUILDDEBUGONSTART=0
+:build
 powershell -NoLogo -NoProfile -ExecutionPolicy ByPass -Command "& """%~dp0eng\common\build.ps1""" -build -restore %*"
+set MSBUILDDEBUGONSTART=%CACHEDMSBUILDDEBUGONSTART%
```

Comment:
I'd put 

```cmd
setlocal
```

at the top of the file (under `@echo off`); then you won't have to worry about resetting state.